addons/stock_account/models/product.py

```diff
@@ -859,12 +859,9 @@ def _get_stock_account_property_field_names(self):
     def _check_valuation_accounts(self):
         fnames = self._get_stock_account_property_field_names()
         for category in self:
-            # "compute" properties in constraint because ORM doesn't support computed properties
-            for property_field in fnames:
-                category[property_field] = category.property_valuation == 'real_time' and (
-                    category[property_field]
-                    or self.env['ir.property']._get(property_field, 'product.category')
-                )
+            if category.property_valuation == 'real_time':
+                if any(not category[account] for account in fnames):
+                    raise ValidationError(_('The stock accounts should be set in order to use the automatic valuation.'))
```

Comment:
Hi, I am not sure if this will cause some bug
https://github.com/odoo/odoo/blob/cb97c0b47cfdb7e668ff058263813e5d56ede330/addons/mrp_account/models/product.py#L128C16-L128C55

when you write `property_stock_account_input_categ_id` to 2 `real_time` records
`record1.property_stock_account_production_cost_id` is empty
`record2.property_stock_account_production_cost_id` is not empty

Then according to the override
`property_stock_account_production_cost_id` will be in the `_get_stock_account_property_field_names` and checked here
as a result, it may raise error here?